### PR TITLE
Address rpmlint warning: devel-file-in-non-devel-package

### PR DESCRIPTION
**Pull request by minghui-liu**
#### Bug/feature Description
rpmlint complains about header files and libraries in non-development packages

#### Affected Platform(s)
* CentOS, openSUSU, RHEL, Fedora

#### Solution Description
Move libraries under /lib and header files under /include into a separate package named `pbspro-devel`

#### Testing logs/output
* [devel-packge-build-and-test-result.txt](https://github.com/PBSPro/pbspro/files/2989901/devel-packge-build-and-test-result.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: pbspro.spec [hunk 1]
```diff
@@ -466,6 +485,10 @@ fi
 %endif
 %exclude %{pbs_prefix}/unsupported/*.pyc
 %exclude %{pbs_prefix}/unsupported/*.pyo
+%exclude %{pbs_prefix}/lib/*.a
+%exclude %{pbs_prefix}/include/*
+%exclude %{pbs_prefix}/lib/*.so*
+%exclude %{pbs_prefix}/lib/pkgconfig/pbs.pc
```
Comment (mike0042) [inline pbspro.spec]: I think you only want to exclude the header files and archive libraries. The shared libraries and pbs.pc should not be excluded. Question: why is it not pbspro.pc?

File: pbspro.spec [hunk 2]
```diff
@@ -505,6 +528,10 @@ fi
 %exclude %{pbs_prefix}/sbin/pbsfs
 %exclude %{pbs_prefix}/unsupported/*.pyc
 %exclude %{pbs_prefix}/unsupported/*.pyo
+%exclude %{pbs_prefix}/lib/*.a
+%exclude %{pbs_prefix}/lib/*.so*
+%exclude %{pbs_prefix}/include/*
+%exclude %{pbs_prefix}/lib/pkgconfig/pbs.pc
```
Comment (mike0042) [inline pbspro.spec]: Same comment as before. Why are these in a different order?

File: pbspro.spec [hunk 3]
```diff
@@ -552,6 +579,19 @@ fi
 %exclude %{pbs_prefix}/unsupported/*.pyc
 %exclude %{pbs_prefix}/unsupported/*.pyo
 %exclude %{_unitdir}/pbs.service
+%exclude %{pbs_prefix}/lib/*.a
+%exclude %{pbs_prefix}/lib/*.so*
+%exclude %{pbs_prefix}/include/*
+%exclude %{pbs_prefix}/lib/pkgconfig/pbs.pc
```
Comment (mike0042) [inline pbspro.spec]: Same comment here.

File: pbspro.spec [hunk 4]
```diff
@@ -552,6 +579,19 @@ fi
...
+%doc README
+%license LICENSE
+
+%files %{pbs_devel}
+%defattr(-,root,root, -)
+%{pbs_prefix}/lib/*.a
+%{pbs_prefix}/lib/*.so*
+%{pbs_prefix}/include/*
+%{pbs_prefix}/lib/pkgconfig/pbs.pc
```
Comment (mike0042) [inline pbspro.spec]: I believe this should only be the archive libraries and header files.

File: pbspro.spec.in [hunk 5]
```diff
@@ -229,6 +230,18 @@ HPC clusters, clouds and supercomputers.
 This package is intended for a client host and provides
 the PBS Professional user commands.
 
+
+%package %{pbs_devel}
+Summary: PBS Professional Development Package
+Group: System Environment/Base
```
Comment (mike0042) [inline pbspro.spec.in]: Should have caught this before. The Group should be "Development/System"

**Comment by minghui-liu** [comment]
Thanks @agrawalravi90 @bayucan and @mike0042.

**Pull request merged**